specs/experimental/inbox-contract.md

```diff
@@ -0,0 +1,113 @@
+# Inbox Contract
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+- [Inbox Contract](#inbox-contract)
+  - [Motivation](#motivation)
+  - [How It Works](#how-it-works)
+  - [Dynamic Updates](#dynamic-updates)
+    - [SystemConfig](#systemconfig)
+      - [setBatchInbox](#setbatchinbox)
+      - [UpdateType](#updatetype)
+    - [How `op-node` knows the canonical batch inbox](#how-op-node-knows-the-canonical-batch-inbox)
+    - [How `op-batcher` knows canonical batch inbox](#how-op-batcher-knows-canonical-batch-inbox)
+  - [Upgrade](#upgrade)
+  - [Reference Implementation](#reference-implementation)
+
+## Motivation
+
+The batch inbox is currently an Externally Owned Account (EOA), which has both advantages and disadvantages:
+
+Advantages:
+- Low submission gas cost due to the absence of onchain execution.
+- Verification logic is moved offchain to the derivation part, protected by a fault dispute game with a correct absolute prestate.
+
+Disadvantage:
+- Onchain verification is not possible.
+
+
+This specification aims to allow the batch inbox to be a contract, enabling customized batch submission conditions such as:
+- Requiring the batch transaction to be signed by a quorum of sequencers in a decentralized sequencing network; or
+- Mandating that the batch transaction call a BLOB storage contract (e.g., EthStorage) with a long-term storage fee, which is then distributed to data nodes that prove BLOB storage over time.
+
+
+## How It Works
+
+The integration process consists of three primary components:
+1. Replacement of the [`BatchInboxAddress`](https://github.com/ethereum-optimism/optimism/blob/db107794c0b755bc38a8c62f11c49320c95c73db/op-chain-ops/genesis/config.go#L77) with an inbox contract: The existing `BatchInboxAddress`, which currently points to an Externally Owned Account (EOA), will be replaced by a smart contract. This new inbox contract will be responsible for verifying and enforcing batch submission conditions.
```

Comment:
The term `replacement` may be misleading as the inbox address can be initialized in the Rollup config (so no replacement is needed).

Reply:
Good point.

Reply:
The term `replacement` is now gone: https://github.com/ethstorage/specs/pull/3/commits/2f22e44289838c26da04bdfb16fc419e8a84b459